src/SeqCli/Cli/Commands/Bench/BenchCommand.cs

```diff
@@ -0,0 +1,213 @@
+        
+        Options.Add(
+            "c|cases=", 
+            @"A JSON file containing the set of cases to run. Defaults to a standard set of cases.", 
+            c => _cases = c);
+
+        _connection = Enable<ConnectionFeature>();
+        _range = Enable<DateRangeFeature>();
+        
+        Options.Add(
+            "reporting-server=", 
+            "The address of a Seq server to send bench results to", 
+            s => _reportingServerUrl = s);
+        Options.Add(
+            "reporting-apikey=", 
+            "The API key to use when connecting to the reporting server", 
+            a => _reportingServerApiKey = a);
+    }
+    
+    protected override async Task<int> Run()
+    {
+        try
+        {
+            var connection = _connectionFactory.Connect(_connection);
+            _reportingLogger = BuildReportingLogger();
+            var cases = ReadCases(_cases);
+            var runId = Guid.NewGuid().ToString("N").Substring(0, 4);
+            var start = _range.Start ?? DateTime.UtcNow.AddDays(-7);
+            var end = _range.End;
+
+            foreach (var c in cases.Cases)
+            {
+                var timings = new BenchCaseTimings();
+                object? lastResult = null;
+                
+                foreach (var i in Enumerable.Range(1, _runs))
+                {
+                    var response = await connection.Data.QueryAsync(
+                        c.Query,
+                        _range.Start ?? DateTime.UtcNow.AddDays(-7),
+                        _range.End,
+                        SignalExpressionPart.FromIntersectedIds(c.Signals)
+                    );
+
+                    timings.PushElapsed(response.Statistics.ElapsedMilliseconds);
+                    var isScalarResult = response.Rows.Length == 1 && response.Rows[0].Length == 1;
+
+                    if (i == _runs && isScalarResult)
+                    {
+                        lastResult = response.Rows[0][0];
+                    }
+                }
+                
+                using (lastResult != null ? LogContext.PushProperty("LastResult", lastResult) : null)
+                using (LogContext.PushProperty("MinElapsed", timings.MinElapsed))
+                using (LogContext.PushProperty("MaxElapsed", timings.MaxElapsed))
+                using (LogContext.PushProperty("Runs", _runs))
+                using (LogContext.PushProperty("Signals", c.Signals))
+                using (LogContext.PushProperty("Start", start))
+                using (LogContext.PushProperty("StandardDeviationElapsed", timings.StandardDeviationElapsed))
+                using (end != null ? LogContext.PushProperty("End", end) : null)
+                using (LogContext.PushProperty("Query", c.Query))
+                {
+                    _reportingLogger.Information(
+                        "Bench run {BenchRunId} for query {Id}. Mean {MeanElapsed:N0} ms with relative dispersion {RelativeStandardDeviationElapsed:N2}", 
```

Comment:
Nit - messages in this codebase are fragments rather than paragraphs or sentences, can just turn the `.` into `:` and lowercase the `mean` :-)